projects/valtimo/dossier/src/lib/services/dossier-column.service.ts

```diff
@@ -57,15 +57,15 @@ export class DossierColumnService {
   }
 
   mapDefinitionColumnsToColumnConfigs(
-    columns: Array<DefinitionColumn>,
-    hasEnvConfig: boolean
+    columns: Array<DefinitionColumn>
   ): Array<ColumnConfig> {
     return columns.map(column => {
       const translationKey = `fieldLabels.${column.translationKey}`;
       const translation = this.translateService.instant(translationKey);
       const validTranslation = translation !== translationKey && translation;
+
       return {
-        key: hasEnvConfig ? column.propertyName : column.translationKey,
```

Comment:
If there is an env config (default definition columns or specific columns for a document definition in environment file), it should use propertyName and if there are API columns it should use translationKey. You just removed logic instead of extending it (hasEnvConfig now only checks for specific columns for a doc def from env file and should be extended to also be `true` when default env columns are used)